lib/glimmer/tk/toplevel_proxy.rb

```diff
@@ -32,19 +32,43 @@ class ToplevelProxy < WidgetProxy
       DEFAULT_HEIGHT = 0
       
       attr_reader :tk
-      attr_accessor :escapable
+      attr_accessor :escapable, :modal, :centered
```

Comment:
I just tested the `modal` behavior. It hides the previous window on every modal window opening. That is not standard behavior. Also, the Mac environment does support real modals as demonstrated in `samples/hello/hello_toplevel.rb` using `mac_style 'utility'`.

Given that the `modal` behavior is not standard, it should be called something else to avoid confusing users. How about calling it `hiding_modal` or something else (if you can't come up with the best name, call it anything, and I could rename it later)?

Also, the real modal behavior on the Mac as demonstrated in Hello, Toplevel, supports pressing ESC button to quickly close the modal and go back to the main window.

I think you should support that behavior with the `hiding_modal` too.

Last but not least, please add examples of using the new `hiding_modal` behavior under `samples/hello/hello_toplevel.rb`

Ideally, all new behavior is added to samples to demonstrate the new features as well (or at least, they should be mentioned in the README.md).

Reply:
Hello Andy!

I'm also sorry for being slow in my responses, the circumstances not always allow me to dedicate as much time for personal projects as I'd like to :(

Yes, you're correct, that's not a standard behavior. May be my choice of the method's name is wrong. How about call it `hide_root`? Because of that's basically what it does - hides the root window when we want to make the user interact with the current toplevel only.

Yes, I tried the approach which also described by the link you provided. And in fact it is still here :) https://github.com/AndyObtiva/glimmer-dsl-tk/pull/14/files#diff-79d73888fd83581749673cadbbcc881b503ad66f694728be37c41415d401210aR55

but I found the root window still exists and the user may switch to it occasionally and lose the toplevel we want them to stay in. I found that may be confusing for the user, that's why I came to this solution.

So, `hide_root`? :) Or `hide_root_window` for more readability? :)

Reply:
Sure, I'll add examples. Sorry, I should have thought about that myself. As soon as I have time :(

Reply:
Do not worry about having the time to respond. I appreciate the time you already put into the PR. And, I am intentionally dividing the PR review into small parts to make it possible for both of us to respond gradually in small replies without having to spend too much time responding to everything at the same time.

Anyways, given that we cannot implement true modality for windows/linux (as per the provided link), then you can keep the attribute name the same (`modal`) for now. 

But, in the future, I will likely make it activate `mac_style 'utility'` on the Mac only to provide true modality on the Mac at least. I think that would be the best compromise. You do not have to make any changes for the `modal` attribute though. I'll merge your work as is for `modal`, and will make the change myself in the future.

Otherwise, do not worry about the examples if it's not a quick task. I could come up with the examples myself and it would be a good exercise for me to better get acquainted with your code changes.